### ISSUE TITLE
Bug: no-multiple-empty-lines ignores failures when there is a single empty line at EOF

### ISSUE THREAD

**Issue by alberto** [eslint/eslint#4214]
I introduced a bug in #4208 

Now when there is a single empty line at the end of the file, no errors are reported.

This should be invalid but it is not:
```js
var a=5;\n\n\n\nvar b = 3;\n
```

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by nzakas** [comment]
I'm not clear on what you're reporting. I thought the point of the earlier change was so that this code wouldn't flagged? Now it seems like you're reporting the reverse is true? Can you explain?

**Comment by alberto** [comment]
The code before the two issues always removed the last empty line. If there were multiple empty lines at the end, it would fail to report that in some cases (depending on max value and the number of empty lines).

Now it only removes the last line when there is exactly one empty line, to avoid reporting when max is 0 (or any value in fact, since it won't report for max > 0).

**Comment by nzakas** [comment]
What do you mean by "removed" the last line? Are you talking about auto-fixing behavior?

**Comment by alberto** [comment]
No, I'm sorry. I am talking about [this line](https://github.com/eslint/eslint/commit/bf4015ccf78e6b31d8c1cb08444abe57403d9487#diff-0f6472c36e2e4d88a28a92f9b5148206R58). I mean excluding it from the checks. Returning is wrong because it prevents other errors from being reported (as the multiple linebreaks in the example above), since this rule is checking the whole file on program exit.

The difference with the original is the condition under which the exclusion happens (only when there is exactly one empty line at the end).

**Comment by nzakas** [comment]
Ah, gotcha!
